src/gsy_e/gsy_e_core/redis_connections/redis_communication.py

```diff
@@ -24,33 +24,41 @@
 
 from gsy_framework.constants_limits import HeartBeat, ConstSettings
 from gsy_framework.exceptions import GSyException
-from gsy_framework.results_validator import results_validator  # NOQA
 from gsy_framework.utils import RepeatingTimer
 from redis import StrictRedis
-from redis.exceptions import ConnectionError
+from redis.exceptions import ConnectionError as RedisConnectionError
 from rq import get_current_job
 from rq.exceptions import NoSuchJobError
 
 import gsy_e.constants
+from gsy_e.gsy_e_core.exceptions import LiveEventException
 
 log = getLogger(__name__)
 
 
-REDIS_URL = os.environ.get('REDIS_URL', 'redis://localhost')
+REDIS_URL = os.environ.get("REDIS_URL", "redis://localhost")
 
 
 if TYPE_CHECKING:
     from gsy_e.models.area import Area
 
 
 class RedisSimulationCommunication:
-    def __init__(self, simulation, simulation_id, live_events):
+    """
+    Handle Redis connection to the simulation, to receive incoming state change messages or live
+    events.
+    """
+    def __init__(self, simulation_status, simulation_id, live_events, progress_info, area):
+        # pylint: disable=too-many-arguments
         self._live_events = live_events
         self._simulation_id = simulation_id if simulation_id is not None else ""
-        self._configuration_id = gsy_e.constants.CONFIGURATION_ID
-        self._simulation = simulation
+        self._simulation_status = simulation_status
+        self._area = area
+        self._progress_info = progress_info
+        # Lambda is needed here in order to avoid storing the area as a member variable.
+        # Instead a closure is used that will capture the area variable in the function scope.
```

Comment:
Sorry, I do not get this comment. Is it maybe outdated and remnant of intermediate state?

Reply:
It is indeed an outdated comment. I tried initially to get rid of the `self._area` member by capturing the constructor value in a closure/lambda, but after seeing that pylint allows us for more member variables that this class had I decided to remove this in order to be easier for the reader. I will remove the comment, thanks!